CellFabric/tests/fabric_CMC_NMOS_v1_default.py

```diff
@@ -67,23 +49,15 @@ def __init__( self, fin_u, fin, finDummy, gate, gateDummy):
                                       clg=UncoloredCenterLineGrid( pitch= p['Fin']['Pitch'], width= p['Fin']['Width'], offset= p['Fin']['Offset']),
                                       spg=SingleGrid( offset=0, pitch=unitCellLength)))
 
+        stoppoint = (gateDummy-1)* p['Poly']['Pitch'] +  p['Poly']['Offset']
+
         self.active = self.addGen( Wire( 'active', 'active', 'h',
                                          clg=UncoloredCenterLineGrid( pitch=activePitch, width=activeWidth, offset=activeOffset),
-                                         spg=CenterLineGrid()))
-        stoppoint = (gateDummy-1)* p['Poly']['Pitch'] +  p['Poly']['Offset']
-        self.active.spg.addCenterLine( 0,         activeWidth_h, False)       
-        self.active.spg.addCenterLine( stoppoint, activeWidth_h,        True)
-        self.active.spg.addCenterLine( unitCellLength-stoppoint, activeWidth_h,         True)
-        self.active.spg.addCenterLine( unitCellLength, activeWidth_h,         False)
+                                         spg=EnclosureGrid( pitch=unitCellLength, offset=0, stoppoint=stoppoint, check=True)))
```

Comment:
Why couldn't we just use an EnclosureGrid here? Does the CenterLine Width have any role to play in a stop point grid?

Reply:
Refactoring the code as I have done still lets the test pass successfully by the way. But it is possible a different configuration will highlight why activeWidth_h is needed.

Reply:
I don't think we want stop point grids to have widths. (Are we using it for anything?) So this must have been a mistake.
There is a difference in the grid numbering here. The enclosure grid has an other construction line at unitCellLength//2, but probably that doesn't matter because you are using major and minor indexing to get at the grid lines.

Reply:
Yes. We were already using minor grid coordinate -1 to get to the legal stop point after unitCellLength//2. So adding an additional grid line made no difference. Also, it didn't look like we were using the stop point width in this primitive at least which is why I asked about any hidden dependencies. 

I will ask @arvuce22 to simplify the other primitives as well based on this discussion. 

Thanks Steve !

Reply:
Thanks, Parijat and Steve! I will update the primitives.

Reply:
Thank you @arvuce22. Do you also have tests for all the primitive generators to make sure I don't accidentally break something? I would like to try & add the FEOL GDS layer numbers to the JSON abstraction so that we can remove the redundant gen_gds_json & gen_lef code.